es6 & beyond/ch4.md

```diff
@@ -306,15 +306,15 @@ function *main() {
 }
 ```
 
-On the surface, this snippet may seem more verbose than the promise chain equivalent in the earlier snippet. However, it offers a much more attractive -- and more importantly, a more understandable and reason-able -- synchronous-looking coding style (with `=` assignment of "return" values, etc.) That's especially true in that `try..catch` error handling can be used across those hidden async boundaries.
+A primeira vista, o trexo de código pode parecer mais verboso que a cadeia de `promise` equivalente no outro trexo de código. Contudo, isso oferece um estilo de código de aparência sincrona muito mais atrativo -- e o mais importante, mais compreensível -- (com a instrução `=` para retornos de valores, etc.) É especialmente verdade que o manipulador de erros `try catch` pode ser usado através dessas fronteiras ocultas assincronas.
 
-Why are we using Promises with the generator? It's certainly possible to do async generator coding without Promises.
+Por que estamos utilizando `Promises` com `generator`? É totalmente possível escrever `generators` assíncronos sem `Promises`.
 
-Promises are a trustable system that uninverts the inversion of control of normal callbacks or thunks (see the *Async & Performance* title of this series). So, combining the trustability of Promises and the synchronicity of code in generators effectively addresses all the major deficiencies of callbacks. Also, utilities like `Promise.all([ .. ])` are a nice, clean way to express concurrency at a generator's single `yield` step.
+`Promises` são sistemas confiáveis que desfazem a inversão de controle de `callbacks` ou `thunks` (veja o título *Async & Performance* dessa série). Então, combinando a confiabilidade das `Promises` e a sincronicidade de código dos `generators` trata efetivamente todas as maiores deficiências das `callbacks`. Também, funções utilitárias como `Promise.all([..])` expressam de forma clara e agradável concorrência em um único passo `yield` dos `generatores`.
 
-So how does this magic work? We're going to need a *runner* that can run our generator, receive a `yield`ed promise, and wire it up to resume the generator with either the fulfillment success value, or throw an error into the generator with the rejection reason.
+Então, como essa mágica funciona? Nós precisaremos de um *runner* que executa nosso `generator`, recebe uma `promise` `yield`, e os liga para formar um `generator` com ambos, o valor de cumprimento, ou lança um erro no `generator` com o motivo da rejeição.
 
-Many async-capable utilities/libraries have such a "runner"; for example, `Q.spawn(..)` and my asynquence's `runner(..)` plug-in. But here's a stand-alone runner to illustrate how the process works:
+Muitos utiliários ou bibliotecas com capacidades assincronas possuiem um tipo de "runner"; por exemplo, `Q.spawn(..)` e <b style="color: red">minha asynquence's `runner(..)` plug-in</b>. Mas aqui está um runner stand-alone ilustrando como o processo funciona:
```

Comment:
Many async-capable utilities/libraries have such a "runner"; for example, `Q.spawn(..)` and my asynquence's `runner(..)` plug-in. But here's a stand-alone runner to illustrate how the process works:

> muitas bibliotecas/utilitarios capazes de rodar o codigo assincrono incluem um _"runner"_; 
 por exemplo, `Q.spawn(..)` e meu plugin the sequencia assincrona `runner(..)`. Mas existe uma versao separada de _runner_ para ilustrar como o processo funciona

desculpe nao peguei o contexto todo, mas deve ser por ai